protocol/messenger_contacts.go

```diff
@@ -1323,7 +1323,14 @@ func (m *Messenger) FetchContact(contactID string, waitForResponse bool) (*Conta
 		WithWaitForResponseOption(waitForResponse),
 	}
 	contact, _, err := m.storeNodeRequestsManager.FetchContact(contactID, options)
-	return contact, err
+	if err != nil {
+		return nil, err
+	}
+
+	if contact == nil {
+		return nil, ErrContactNotFound
```

Comment:
I wonder if this is the right solution 🤔 

The idea of returning `nil` was to explicitly show that there was no error during requests. Just the contact not found. So both return args were `nil`.

Yes, we use `ErrContactNotFound` in some places. Just to prevent checking both `err` and `contact`. I'm personally not both hands on this idea. At some upper levels maybe yes, but not on lower levels.

I believe right now all of the usages of `FetchContact` and `FetchCommunity` expect `nil, nil` to be returned when the content wasn't found. It's noted in the function description:
https://github.com/status-im/status-go/blob/b1433e6bfa8406fb31017e86be53bf8dca12d439/protocol/messenger_store_node_request_manager.go#L76

So maybe in this case we could just check the returned value in `UnfurlURLs` to prevent regressions?

Reply:
Thanks for clarification! I'll fix it

Reply:
done